control_panel_api/models.py

```diff
@@ -32,24 +31,10 @@ class Meta:
 
 
 class S3Bucket(TimeStampedModel):
-    # See: AWS' Bucket Restrictions and Limitations
-    # http://docs.aws.amazon.com/en_gb/AmazonS3/latest/dev/BucketRestrictions.html
-    #
-    # A label starts with a letter (preventing labels starting with digits to
-    # avoid IP-like names) and ends with a letter or a digit. It has 0+
-    # letters, digits or hyphens in the middle
-    #
-    # An S3 bucket name starts with a label, it can have more than one label
-    # separated by a dot.
-    LABELS_REGEX = '^([a-z][a-z0-9-]*[a-z0-9])(.[a-z][a-z0-9-]*[a-z0-9])*$'
-    # An S3 bucket name needs to be min 3 chars, max 63 chars long.
-    LENGTH_REGEX = '^.{3,63}$'
-
     name = models.CharField(unique=True, max_length=63, validators=[
-        RegexValidator(regex=LENGTH_REGEX,
-                       message="must be between 3 and 63 characters"),
-        RegexValidator(regex=LABELS_REGEX,
-                       message="is invalid, check AWS S3 bucket names restrictions (for example, can only contains letters, digits, dots and hyphens)"),
+        validators.validate_env_prefix,
+        validators.validate_s3_bucket_length,
+        validators.validate_s3_bucket_label,
```

Comment:
I know the name of this wasn't probably great in the first place but I'd probably call this `validators.validate_s3_bucket_labels` (plural) as the idea was for this regex to validate that the S3 bucket name was 1+ labels (as defined by AWS S3 bucket naming restrictions document).

Reply:
Done